packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -5875,4 +5875,38 @@ describe("Teal Opcodes", function () {
       });
     });
   });
+
+  describe("BitLen opcode", function () {
+    let stack: Stack<StackElem>;
+    let interpreter: Interpreter;
+    this.beforeEach(() => {
+      stack = new Stack();
+      interpreter = new Interpreter();
+    });
+
+    it("should work with number", () => {
+      stack.push(8n);
+      const op = new BitLen([], 1, interpreter);
+      op.execute(stack);
+      assert.equal(stack.pop(), 4n);
+    });
+
+    it("shoud work with a shore byte array", () => {
+      const byte = "abcd";
+      const op = new BitLen([], 1, interpreter);
+
+      stack.push(parsing.stringToBytes(byte));
+      op.execute(stack);
+      assert.equal(stack.pop(), 31n);
+    });
+
+    it("shoud work with a long byte array", () => {
+      const byte = "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+      const op = new BitLen([], 1, interpreter);
+
+      stack.push(parsing.stringToBytes(byte));
+      op.execute(stack);
+      assert.equal(stack.pop(), 623n);
+    });
```

Comment:
Where are these tests referenced from? We need to add few corner cases as well. You can reference `go-algorand` tests.